bootloader/grub.go

```diff
@@ -343,11 +344,25 @@ func (g *grub) UpdateBootConfig() (bool, error) {
 	// XXX: do we need to take opts here?
 	bootScriptName := "grub.cfg"
 	currentBootConfig := filepath.Join(g.dir(), "grub.cfg")
-	if g.recovery {
+	if g.role == RoleRecovery {
 		// use the recovery asset when asked to do so
 		bootScriptName = "grub-recovery.cfg"
 	}
-	return genericUpdateBootConfigFromAssets(currentBootConfig, bootScriptName)
+	bootUpdated, err := genericUpdateBootConfigFromAssets(currentBootConfig, bootScriptName)
+	if err != nil {
+		return false, err
+	}
+	recovUpdated := false
+	if g.role == RoleRunMode {
+		// TODO: How to find out ubuntu-seed folder?
+		recovUpdated, err = genericUpdateBootConfigFromAssets(
```

Comment:
I don't think the run mode bootloader wrapper should implicitly try to update the recovery one. This would be done by the caller, by first creating the instance of the right bootloader and then invoking UpdateBootConfig(), iow the policy for update would be implemented higher up the stack. There's a TODO about updating recovery in overlord/devicestate/handlers_bootconfig.go where this policy is implemented.

Reply:
Right, thanks for the hint. Anyway the PR is not bumping the edition number in the end, so I have removed this part as it is not needed for the moment.